src/lib/components/ui/character-tabs/talent-table.svelte

```diff
@@ -0,0 +1,115 @@
+<script lang="ts">
+	import Card from '$lib/components/ui/card/card.svelte';
+	import Text from '$lib/components/typography/Text.svelte';
+	import * as Table from './table';
+	import * as Collapsible from '$lib/components/ui/collapsible';
+	import { Slider } from '$lib/components/ui/slider';
+	import type { Passive, Skill } from '$lib/types/data/Character';
+	import Button from '$lib/components/ui/button/button.svelte';
+	import Icon from '../icon/icon.svelte';
+	import { mdiClose } from '@mdi/js';
+
+	export let skills: Skill[];
+	export let passives: Passive[];
+	const skillTypes = ['Normal Attack', 'Elemental Skill', 'Elemental Burst'];
+
+	$: talentLevels = [9, 9, 9];
+	$: detailOpen = false;
+	$: currentSkill = 0;
+</script>
+
+<div
+	data-open={detailOpen}
+	class="data-[open=true]:hidden flex flex-col gap-4 h-full flex-shrink lg:overflow-auto"
+>
+	<div class="flex flex-col gap-2 h-full">
+		{#each skills as skill, i}
+			<Button
+				variant="ghost"
+				class="h-full flex flex-col items-start bg-tertiary"
+				on:click={() => {
+					currentSkill = i;
+					detailOpen = true;
+				}}
+			>
+				<Text type="h4" class="text-wrap text-left">{skill.name}</Text>
+				<Text type="p" class="text-base">{skillTypes[i]}</Text>
+			</Button>
+		{/each}
+	</div>
+	<div class="flex flex-1 flex-col gap-2 lg:flex-row lg:max-h-full">
+		{#each passives as passive}
+			<Card class="flex flex-col bg-neutral border-secondary w-full lg:overflow-auto">
+				<Collapsible.Root>
+					<Collapsible.Trigger>
+						<Text type="p" class="font-bold text-lg">
+							{`${passive.level == 0 ? 'Utility Passive: ' : `A${passive.level}: `} ${passive.name}`}
+						</Text>
+					</Collapsible.Trigger>
```

Comment:
The trigger should grow. Otherwise opening the collapsible is unreliable for some passives

Reply:
What do you mean by this? Would changing flex-row to flex-col be better for the ui overall or should I still try to keep the passives in a single row?

Reply:
![Screenshot 2024-04-06 152238](https://github.com/dval-in/dvalin-frontend/assets/14811944/156d228c-82b2-4d61-a945-ef7a496c61ee)

The collapsible is only triggered when clicking on the text area. This makes it hard to open when it doesnt fill the whole space